### ISSUE TITLE
Distribute BTC fees to contributors of last DAO cycle

### ISSUE THREAD

**Issue by chimp1984** [bisq-network/proposals#305]
> _This is a Bisq Network proposal. Please familiarize yourself with the [submission and review process](https://docs.bisq.network/proposals.html)._

An alternative idea to previously discussed ideas how we can improve the handling of the BTC fees would be to distribute the BTC fee to all contributors of the last DAO cycle (or a collection of the past cycles) weighted by the received compensation.

It would have the positive psychological effect that every contributor receives a stream of payments every day. But a problem would be that many tiny payments blow up the BitcoinJ wallet in Bisq. To avoid that we could advice the contributors to use a non-Bisq wallet for that (e.g. Bitcoin core). We would need to extend the Compensation request data to include a BTC donation address. We have the `extraDataMap` field for that so technically that should work.

This would solve the problem of the BTC fee sent to the burningman and getting burned. The loss for the DAO from the missing burned BSQ would be compensated by lower issuance for compensation requests. E.g. If it shows that contributor get about 20% of their requests extra by that fee we should get to lower amounts for future requests to not overpay contributors.

I think that could be implemented rather easy. It is just a lookup to the compensation requests and collect those data and make a random selection and it requires to add that extra field into the comp. request UI.

**Comment by ripcurlx** [comment]
So it wouldn't be handled by the filter, but rather each node takes the value from the DAO data? If we do this small txs we would need to take into account the USD value of each tx at the certain time to have a correct USD amount to deduct from any newly issued CR, right? What about DAO contributors from last cycle that don't contribute in the current cycle or completely stop contributing? They would still earn BTC from fees. It does sound interesting to get rid of this BTC <> BSQ conversion by the burning man. Maybe we should also have a look why traders prefer paying higher BTC fees than using BSQ in the first place. Maybe there is a more easy win to gain by improving these issues.

**Comment by ripcurlx** [comment]
Maybe one way could be that contributors burn BSQ upfront and get future BTC fees based on the USD value of the burnt BSQ at that point of time up to this Dollar amount. Similar as we do it with the security incident victims.

**Comment by chimp1984** [comment]
> What about DAO contributors from last cycle that don't contribute in the current cycle or completely stop contributing? They would still earn BTC from fees.

It would only work once we have added the BTC address. How we apply it to past cycles is open but with BSQ it is also the case that part contributors benefit from a rising BSQ price. But to apply some decay function simliar we do with merits decaying over 2 years to 0, might be justified. To have it more aggressive so most will be distributed to recent contributions creates more incentives to contribute on Bisq, so might be not that bad as well. Also it only covers a part of the revenue.
As we still do not have clear numbers it is hard to estimate but from some calculations I did over the past days It seems to me that only 25% of the trade fees are paid in BSQ and 75% in BTC. But I need more time to be sure about those numbers.

> Maybe we should also have a look why traders prefer paying higher BTC fees
Yes I agree that we should find out why that is the case. My interpretation is that trade fees are not that high in BTC that users really care about it or are willing to do the extra work to get BSQ. So we can increase BTC fees to widen the discount and make it easier/more attractive to buy BSQ on the market.

One could argue that we do not need BSQ anymore if we distribute 100% BTC fee (e.g. remove option to use BSQ for fees), but then the value of BSQ would go down as there is no demand and with that the voting would become insecure and with that the compensation requests and by the the BTC distribution algorithm. So we depend on BSQ for that solution. We just can use it for a certain portion, but even if the portion is too high it will lower the security of the overall system, so we should be cautious to see this as an alternative for BSQ fee, but rather as a solution to avoid the burningman role, but still work hard to give BSQ fee the 70-80% share as it was intended. If my above numbers are correct we are far away from that.  

> If we do this small txs we would need to take into account the USD value of each tx at the certain time to have a correct USD amount to deduct from any newly issued CR

I have not considered it to be "exact" and be used for accounting that way. I think it should be enough if we see the % of what was paid in BTC fees and use that as guidance for the next cycle to reduce it from the USD requests. But how to do that exactly is up for discussion, but as it happens delayed after the request I think there will be no perfect solution and a one time contributor who had luck in a month of super high volume and not continue afterwards might benefit from it. But I think that is not a showstopper and BSQ/BTC volatility are other risks everyone in the space has to deal with. 

> Maybe one way could be that contributors burn BSQ upfront and get future BTC fees based on the USD value of the burnt BSQ at that point of time up to this Dollar amount. Similar as we do it with the security incident victims.

I think that would be more complicate technically and conceptually.

What I like in that is that it removes the need for the burningman at least for one part of his role. As the BM is a problematic and hard to solve problem the best solution for such circumstances is to get rid of it completely...

**Comment by MwithM** [comment]
> What I like on that proposal is that it removes the need for the burningman at least for one part of his role. As the BM is a problematic and hard to solve problem the best solution for such circumstances is to get rid of it completely...

The filter would not remove the need to depend on some anonymous contributor(s) taking care of the donation address.
I find the tradebot (#304) and swaps in general more promising (although they don't get rid of burningman anyway). At this proposal, I find it will be hard to follow spending. It will arise disputes between contributors about who got more BTC, as being compeansated in BTC is more desirable than in BSQ.
The distribution is more decentralized when the burningman burns BSQ bought to traders than using the filter, reducing legal risks.

**Comment by chimp1984** [comment]
> > What I like on that proposal is that it removes the need for the burningman at least for one part of his role. As the BM is a problematic and hard to solve problem the best solution for such circumstances is to get rid of it completely...
> 
> The filter would not remove the need to depend on some anonymous contributor(s) taking care of the donation address.
> I find the tradebot (#304) and swaps in general more promising (although they don't get rid of burningman anyway). At this proposal, I find it will be hard to follow spending. It will arise disputes between contributors about who got more BTC, as being compeansated in BTC is more desirable than in BSQ.
> The distribution is more decentralized when the burningman burns BSQ bought to traders than using the filter, reducing legal risks.

It is not using the filter but DAO data from compensation requests.
Yes it only solves one of 2 parts of the BM but at least that would be solved by that.
It is more decentralized and much easier to implement than the trade bot or swap ideas.
The distribution is statisitcal so over longer period its evenly distributed according to BSQ amounts in comp. requests.
I think the only issue is that it changes the revenue distributions to benefit more recent contributors, but that can be seen positive from Bisq perspective to creat more incentive to contribute now. For past contributors who are not active anymore it has a disadvantage. But to evenly benefit old inactive contributors to new ones was never an intendted policy but a result of how the BSQ burning works. I think to benefit new active contributors is a better policy. It also does not change 100% but only the part which is spent in BTC fees. Also as stated above we should continue to improve incentives that users use BSQ, so the goal to get 70-80% on fees in BSQ is still valid and by that the effect to old inactive contributor is lower.

**Comment by pazza83** [comment]
Thanks for the proposal.

My initial thoughts were this sounds good. Getting paid in BTC would be nice.

After a little think though I am not in favor based on the fundamental principle that I believe that Bisq contributors should be aligned to the success of Bisq.

Bisq paying contributors in BSQ is a fantastic way to align all contributors with Bisq's long term success.

I would worry that paying contributors in BTC would have a negative impact on this alignment as it takes away their 'stake' in Bisq.

I see BSQ as sweat equity, essentially a stake in the organization. Paying contributors in BSQ gives them this stake in Bisq.

I am in favor of addressing issues with the Burningman but I do not think this is the best long term solution for Bisq.

**Comment by chimp1984** [comment]
@pazza83 
I understand and share your concerns.
In case we do not find a solution for the other part of the BM (funds delayed payout tx) this proposal will not help much. But if we find a soltution for that we can get rid of the BM and then I think this proposal is worth it. The biggest problem in the DAO concept is the BM.
Also we have to be clear that BSQ fee payment still need to be (become) the major part of the fee payment, the 70-80% share is still our target and he have to find more agressive means to get there (e.g. increase the discount). As stated above the DAO and that distribution would not work if 100% would be in BTC.
So I think that the bigger part of contributors earning would be still in BSQ.
   
Also we should keep in mind the positive psychological factor to receive a permanent payment stream. I heard from previous arbitrators who received the trade fee back then, that it was a very motivating feeling even the amount was not that high.
With BSQ we might have a bit the opposite psychological feeling. It is a bit hard to sell due the low liquidity and some contributors (or new ones) might not take it serious as real money but rather as a nice to have speculative share.

**Comment by pazza83** [comment]
What are the reasons why people choose to pay in BTC over BSQ?

Would there be a big pushback if trade fees had to be paid in BSQ? 

If it was technically possible and wasn't too controversial, maybe the DAO could even airdrop a small amount of BSQ to existing users Bisq BTC addresses wallets based on their unique onion addresses to get them started!

**Comment by chimp1984** [comment]
> What are the reasons why people choose to pay in BTC over BSQ?

I don't know but I think the extra work to get BSQ is probably the main reason. Maybe also lack of knowledge about it.

> Would there be a big pushback if trade fees had to be paid in BSQ?

That would not work for first time traders as they need to buy BSQ and pay a fee. I also fear it would cause too much restriction when people felt forced to get some token.

> If it was technically possible and wasn't too controversial, maybe the DAO could even airdrop a small amount of BSQ to existing users Bisq BTC addresses wallets based on their unique onion addresses to get them started!

No that would not work as it need to be sent to a BSQ address. And I mad negative experience whith giving thing away for free. Most people consider it to has no value then.

**Comment by chimp1984** [comment]
> What are the reasons why people choose to pay in BTC over BSQ?

I don't know but I think the extra work to get BSQ is probably the main reason. Maybe also lack of knowledge about it.

> Would there be a big pushback if trade fees had to be paid in BSQ?

That would not work for first time traders as they need to buy BSQ and pay a fee. I also fear it would cause too much restriction when people felt forced to get some token.

> If it was technically possible and wasn't too controversial, maybe the DAO could even airdrop a small amount of BSQ to existing users Bisq BTC addresses wallets based on their unique onion addresses to get them started!

No that would not work as it need to be sent to a BSQ address. And I mad negative experience whith giving thing away for free. Most people consider it to has no value then.

**Comment by pazza83** [comment]
Ok, maybe the [new GUI ](https://github.com/bisq-network/projects/issues/47) could be developed to promote / onboard users to using BSQ for trade fees.

**Comment by cbeams** [comment]
The proposal is a clever solution to how to equitably distribute btc fees, but I’m concerned it misaligns incentives. If I’m getting paid directly in btc, I don’t have much incentive to make it easy and attractive for users to acquire and use BSQ. If we were already in a place where btc fees were a small fraction of overall revenues, I’d be all for this. I think it might be premature to do it now.

**Comment by chimp1984** [comment]
> The proposal is a clever solution to how to equitably distribute btc fees, but I’m concerned it misaligns incentives. If I’m getting paid directly in btc, I don’t have much incentive to make it easy and attractive for users to acquire and use BSQ. If we were already in a place where btc fees were a small fraction of overall revenues, I’d be all for this. I think it might be premature to do it now.

I agree with your concern. But if the incentives are right now (they are I think) what explains the fact that after nearly 2 years DAO we have no clue about our revenue/finance? 

Nobody knows how much BSQ vs BTC fees are used. We even don't really know whats the real % we receive from trade volume. All we have is a guesstimate. If even contributors have no clue one can imagine why there is basically zero interest from people to consider BSQ as equity from an investment point of view. 

And commitment and motivation is sub-optimal in Bisq. The lack that this above major failure has not been resolved for so long is a clear evidence. Further there are many more open issues (and many things which have been solved over the past months have been pretty easy low hanging fruits) which have been dangling around since years as well. 

So my conclusion is that Bisq lacks of incentives (luckily it has improved a lot over that past weeks/months and with 2 new potential fulltime devs we are getting much better) and most contributors are treating it as hobby project on the side and the BSQ payment is not taken sereiously like a payment. 
And from my own experience when selling BSQ I have to confirm that the effort to sell larger amounts (due low liquidity) lead to the position to keep it as speculative long term investment but it does not feel like getting a real payment.

So my conclusion is the incentives are pretty bad as it is in the moment, not from theory but from observed reality.

Adding a part as BTC payment stream would increase those incentives IMO. I guess that many contributors who suddenly receive next to their 5000 BSQ payment a steady stream of maybe 0.1 BTC over a month will get much more motivation that this is something real today and not some potentially valuable equity in the future.
It would also reduce sell pressure for those who need to pay their bills as they might be able to cover that from the BTC part. 

But as stated above we have to be clear that this never must become worse at it is today (if my numbers are right we only have 25% of fees paid in BSQ). But I think there are enough stewards in the poroject who understand the reasons and have enough stake to take care of that.

**Comment by jmacxx** [comment]
>Adding a part as BTC payment stream would increase those incentives IMO. I guess that many contributors who suddenly receive next to their 5000 BSQ payment a steady stream of maybe 0.1 BTC over a month will get much more motivation that this is something real today and not some potentially valuable equity in the future.

Sorry, but I disagree strongly with this and would like to share my opinion:

- BSQ is money, just with different fluctuations than BTC or USD.  
- The existing method of using BSQ for fees and compensation (& governance) is vastly better than this proposal.

As a recent full time contributor, getting paid for actual deliverables is more motivational than just keeping a seat warm.  
BSQ is what makes this project cool, its use case is a unique feature in the OSS world.

**Comment by chimp1984** [comment]
@jmacxx Great to see that support for BSQ and the DAO! I totally share and to be honest I am surprised that it did not get more traction and associated value (price is not rising but was long time very undervalued). 

The primary motivation of that proposal was to get rid of the burning man at least for one funtion he exercises.
As it seems the proposal does not get sufficient support I think we can consider it as "stalled" for now and once we find a solution for the other function of the BM (burning delayed payout tx funds) we can reconsider that one as well.

**Comment by cbeams** [comment]
It’s good to hear the different opinions on this. I’d be happy to hear from others before shelving this completely.

**Comment by Conza88** [comment]
> Maybe we should also have a look why traders prefer paying higher BTC fees than using BSQ in the first place. Maybe there is a more easy win to gain by improving these issues.

**There is:** 
- [Improve BSQ account creation & purchase process](https://github.com/bisq-network/bisq/issues/5046) #5046
- [Add 'low priority' payment account type to BSQ ](https://github.com/bisq-network/bisq/issues/5153 )#5153
- [Add "Buy BSQ" button next to trade fee selector](https://github.com/bisq-network/bisq/pull/5055) #5055

I've also recently asked whether BSQ should be considered an 'altcoin' (as it currently dilutes the BUY BTC field, with SELL BSQ options...), as well as in the OFFERS BY PAYMENT section - altcoins on top (in reality, 56 of those are BSQ)... 

![image](https://user-images.githubusercontent.com/51952482/106971973-0b7d7180-67a4-11eb-94cc-c3684ee47939.png)

I don't see it as an altcoin. Not sure how it should be treated, but think separated from "BUY BTC" for sure... and have its own category in Offers By payment. 

It's the lifeblood of the DAO, different to a shitty altcoin, and should be treated as such imo. 

Whether that's having a special BSQ market place in-app, I don't know. But certainly all of the above would go a long way to increasing the value of BSQ (lowering all the crazy barriers currently in place for someone new to purchase!)

**Comment by RefundAgent** [comment]
Thinking about this proposal I'm lately quite in favor from initial scepticism. This makes Bisq more like a corporation and the stream of BTC is a salary. BSQ is the share and is illiquid because the market cap of Bisq is low. BSQ will only increase in price and trading activity if investors believe Bisq will grow its revenue. Nothing else matters for them. Nothing else.

People don't care about paying their fees in BTC, the fee is certainly not what is limiting usage of Bisq. The only way to incentivize people to use BSQ for the trading fee is to increase the BTC-fee. Certainly not by decreasing the BSQ-fee.
Combining this proposal with a bot that buys BSQ to burn would reduce both the amount of BTC held at central points of control and take away the BM. The bot will also buy smaller amounts of BSQ many times which will improve price discovery on the BSQ-market. It is still not perfect but getting better.

**Comment by Conza88** [comment]
> The only way to incentivize people to use BSQ for the trading fee is to increase the BTC-fee. Certainly not by decreasing the BSQ-fee.

Yeah, that is categorically _**wrong**_... especially given the purchase experience is so undeniably _**woeful**_ from a CX perspective.

Make it [easier to buy BSQ](https://github.com/bisq-network/proposals/issues/305#issuecomment-773692331), it's not rocket science folks.

**Comment by sqrrm** [comment]
I don't think this proposal is moving in the right direction. The DAO and BSQ was set up to avoid exactly this kind of solution.

Short term it's some dev work, added risks of bugs and harder accounting. Longer term mining fees will likely make this scheme unviable anyway and we need to find some other fee payment solution. It also doesn't get rid of the burningman so all in all, I don't think this is a good proposal.

**Comment by Mycelial1** [comment]
This thread turned into discussion about how we could get users to burn BSQ, and create more interest in BSQ generally.

So I created a mockup that was inspired by the ethereum DEXes like Uniswap and 1inch. It's just an icon that displays your current BSQ balance.

![samplebsqcounter](https://user-images.githubusercontent.com/34950746/108474432-3b1c8700-7244-11eb-9706-201b46d7846f.PNG)

![croppedsamplebsqcounter](https://user-images.githubusercontent.com/34950746/108474408-30fa8880-7244-11eb-9b80-bdf5e3908482.png)

I think BSQ just needs to be front and center in the UI. Period. I have felt this way since the DAO was launched.

> If it was technically possible and wasn't too controversial, maybe the DAO could even airdrop a small amount of BSQ to existing users Bisq BTC addresses wallets based on their unique onion addresses to get them started!

I saw Pazza's comment AFTER I created this mockup.

 These Automated Market Makers like Uniswap and 1inch airdropped their own governance tokens recursively to users that helped bootstrap the DEX and they added a GIANT icon at the top of the UI that displays your eligible claim/current balance.

It has now become a trend with these AMM clones to do their own airdrops to incentivize people to be users and liquidity providers. 

I'm not saying we have to do an airdrop (though that would be cool), but I wanted to point out that it works for creating awareness and buzz.

Regardless, I think BSQ just needs to be front and center in the UI. A simple icon that displays your balance would be WAY MORE advertisement for BSQ than having to search for it as a regular altcoin like it is now.
